src/classes/Entity/types.ts

```diff
@@ -227,7 +227,8 @@ export type InferItemAttributeValue<
     | GSISortKeyDefinition
     | PureAttributeDefinition
     ? Definition['type'] extends DynamoDBTypes
-      ? FromDynamoData<A.Cast<Definition['type'], DynamoDBTypes>>
+      ? Definition['setType'] extends DynamoDBKeyTypes
+       ? FromDynamoData<Exclude<Definition['setType'], undefined>>[] : FromDynamoData<A.Cast<Definition['type'], DynamoDBTypes>>
```

Comment:
Your can write `NonNullable<Definition['setType']>`, it's equivalent and a bit shorter

Reply:
🔥